pkg/sql/show_tenant.go

```diff
@@ -124,89 +152,103 @@ func (n *showTenantNode) initReplicationStats(params runParams, job *jobs.Job) e
 			ptp := params.p.execCfg.ProtectedTimestampProvider
 			record, err := ptp.GetRecord(params.ctx, params.p.Txn(), *stats.IngestionDetails.ProtectedTimestampRecordID)
 			if err != nil {
-				return err
+				// Protected timestamp might not be set yet, no need to fail.
+				log.Warningf(params.ctx, "protected timestamp unavailable for tenant %q and job %d: %v",
+					details.DestinationTenantName, job.ID(), err)
+				return stats, nil, nil
 			}
-			n.protectedTimestamp = record.Timestamp
+			protectedTimestamp = record.Timestamp
 		}
 	}
-	return nil
+	return stats, &protectedTimestamp, nil
 }
 
-func (n *showTenantNode) startExec(params runParams) error {
-	tenantName, err := n.getTenantName(params)
-	if err != nil {
-		return err
-	}
+func (n *showTenantNode) getTenantValues(
+	params runParams, tenantName roachpb.TenantName,
+) (*tenantValues, error) {
 	tenantRecord, err := GetTenantRecordByName(params.ctx, params.p.execCfg, params.p.Txn(), tenantName)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	n.tenantInfo = tenantRecord
-	jobId := n.tenantInfo.TenantReplicationJobID
+	var values tenantValues
+	values.tenantInfo = tenantRecord
+	jobId := values.tenantInfo.TenantReplicationJobID
 	if jobId == 0 {
 		// No replication job, this is a non-replicating tenant.
 		if n.withReplication {
-			return errors.Newf("tenant %q does not have an active replication job", tenantName)
+			return nil, errors.Newf("tenant %q is not a replicated tenant", tenantName)
 		}
-		n.tenantStatus = tenantStatus(n.tenantInfo.State.String())
-		return nil
+		values.tenantStatus = tenantStatus(values.tenantInfo.State.String())
+		return &values, nil
 	}
 
-	switch n.tenantInfo.State {
+	switch values.tenantInfo.State {
 	case descpb.TenantInfo_ADD:
 		// There is a replication job, we need to get the job info and the
 		// replication stats in order to generate the exact tenant status.
 		registry := params.p.execCfg.JobRegistry
 		job, err := registry.LoadJobWithTxn(params.ctx, jobId, params.p.Txn())
 		if err != nil {
-			return err
+			values.tenantStatus = tenantStatus(fmt.Sprintf(string(replicationUnknownFormat), err))
```

Comment:
It makes me a little nervous that we aren't failing loudly enough here. Should we log `ERROR: %s` instead of `REPLICATION UNKNOWN`, here and below?

Reply:
Agree. If we have a non-zero jobID, that job ID should have a status we should be able to look up. This presents a non-trivial error I think.